site/content/blog/carvelizing-helm-chart.md

```diff
@@ -0,0 +1,455 @@
+
+Carvel is a collection of small sharp tools that breaks up the problem into smaller building blocks with clean boundaries. This approach facilitates interoperability with other tools, giving users the option to swap out pieces with tools of their choice. Carvel declarative APIs approach provides a familiar experience for Kubernetes users. This means that package management using Carvel is about declaring desired state via a Package CR and have it be continuously reconciled to the desired state on a Kubernetes cluster.
+
+## Prerequisites
+
+Basic knowledge of imgpkg, kbld, kapp-controller, kapp
+
+[`kbld`](https://carvel.dev/kbld/): kbld incorporates image building and image pushing into your development and deployment workflows.
+
+[`imgpkg`](https://carvel.dev/imgpkg/): A tool to package, distribute and relocate your Kubernetes configuration and dependent OCI images as one OCI artifact: a bundle.
+
+[`kapp`](https://carvel.dev/kapp/): kapp is a cli used to deploy and view groups of Kubernetes resources as “application”.
+
+[`kapp-controller`](https://carvel.dev/kapp-controller/): kapp-controller provides declarative APIs to create, customize, install, and update your Kubernetes applications into packages. It also continuously fetches and reconciles resources to converge into their desired state.
+
+**Installing Carvel Tools**
+
+We'll be using Carvel tools throughout this tutorial, so first we'll install them using [`install.sh`](https://carvel.dev/install.sh) script.
+
+```bash
+$ wget -O- https://carvel.dev/install.sh > install.sh
+
+# Inspect install.sh before running...
+$ sudo bash install.sh
+# Install kapp-controller
+$ kubectl apply -f https://github.com/vmware-tanzu/carvel-kapp-controller/releases/latest/download/release.yml
+```
+
+--------------------------
+
+## Authoring a Carvel Package
+
+To create a package, we need to create two Custom Resources (CRs). We will go through step by step to author an nginx helm chart:
+
+**1. Create PackageMetadata CR**: Package Metadata contains high level information and description about the package. Multiple versions of a package share same package metadata.
+
+Save the below PackageMetadata CR to a file named `pkgMetadata.yaml`
+```yaml
+
+apiVersion: data.packaging.carvel.dev/v1alpha1
+kind: PackageMetadata
+metadata:
+  # This will be the name of our package metadata
+  name: nginx.bitnami.vmware.com
+spec:
+  displayName: "Bitnami Nginx Carvel Package"
+  longDescription: "Proxifying Server"
+  shortDescription: "Proxifying Server"
+  categories:
+  - proxy-server
+  providerName: VMWare
+  maintainers:
+  - name: "Carvel"
+  - name: "CarvelInd"
+  - name: "Rohit Aggarwal"
+```
+
+**2. Package Helm Chart**: The Package CR is config including metadata and references to OCI artifacts that informs the package manager what software it holds and how to install itself onto a Kubernetes cluster.
```

Comment:
This seems to indicate that OCI artifacts  instruct the Package about the installation process
We could add more clarity by saying
"The Package CR manifest includes metadata and a specification which points to OCI artifacts and informs the package manager how to process and install the required resources on the cluster"
The OCI artifact will define the resources that are installed though, I wonder if that should be captured.